File: 20190519python/0747. Largest Number At Least Twice of Others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        if len(nums)==1:
            return 0
        FN,SN=0,1
        if nums[0]<nums[1]:
            FN=1
            SN=0
        
        for k in range(2,len(nums)):
            if nums[k]>nums[FN]:
                FN,SN=k,FN
            elif nums[k]>nums[SN]:
                SN=k
        
        if nums[FN]>=nums[SN]*2:
            return FN
        return -1

File: archives/C#/0747. Largest Number At Least Twice of Others.cs
public class Solution {
    public int DominantIndex(int[] nums) {
        int first=int.MinValue;
        int second=int.MinValue;
        int loc=0;
        for (int i=0;i<nums.Length;i++){
            if (nums[i]>first){
                second=first;
                first=nums[i];
                loc=i;
            }
            else if (nums[i]<first && nums[i]>second){
                second=nums[i];
            }
        }
        if(second==int.MinValue){
            return 0;
        }
        if(first>=second*2){
            return loc;
        }
        else{
            return -1;
        }
    }
}
